### ISSUE TITLE
Amesos: mumps solver gives segfault during SymbolicFactorization()

### ISSUE THREAD

**Issue by swiesheier** [trilinos/Trilinos#12647]
When solving a small linear system (~8000 degrees of freedom), I produce the following stack trace

![mumps](https://github.com/trilinos/Trilinos/assets/83650176/35cc2408-0d76-48db-bddb-34d378855f7b)

 `dmumps_c(&(MDS));` is referred to as "initialization of MUMPS" . 
That said, I am not sure whether my observation is a bug or not. I can factorize my matrix using Superlu_Dist or any serial package such as UMFPACK. 
Interestingly, even smaller problems (~1000 degrees of freedom) can be factorized with MUMPS. 

Any ideas?

**Comment by cgcgcg** [comment]
@trilinos/amesos

**Comment by cgcgcg** [comment]
@swiesheier Could you double check that the matrix is not singular?

**Comment by swiesheier** [comment]
@cgcgcg So likely a bug in metis/gmsh ? 

Anything I can do to narrow down the problem? The gmsh library does not provide debug information I think. 

Strange is that it works for small problems (~1,000 degrees of freedom).

**Comment by cgcgcg** [comment]
You could try to reproduce the bug with MUMPS directly to be sure.

**Comment by swiesheier** [comment]
What do you mean? Making the call `dmumps_c(...)` directly without using trilinos?

**Comment by cgcgcg** [comment]
Right. But let's see if someone from the @trilinos/amesos team has more insight.

**Comment by cgcgcg** [comment]
@ndellingwood @iyamazaki Do you have any idea what could be happening here?

**Comment by iyamazaki** [comment]
I agree with @cgcgcg. The issue may not be the Amesos MUMPS interface, but may be with your MUMPS to METIS interfacing. It may be good to run MUMPS' example/test with the same matrix.

**Comment by swiesheier** [comment]
@iyamazaki 

`MUMPS' example/test`

Can you point me to that folder?
They have no github repo if I am not mistaken.

**Comment by iyamazaki** [comment]
Under MUMPS, I see a directory called ``examples``, @swiesheier .

**Comment by swiesheier** [comment]
I can only access the include directory on my installation. On their [website](http://mumps-solver.org/index.php?page=dwnld), I also see no source code. Can you give me a link, @iyamazaki

**Comment by iyamazaki** [comment]
Under my MUMPS directory, I have:
````
$ ls MUMPS_5.3.3
ChangeLog  CREDITS  doc  examples  include  INSTALL  lib  libseq  LICENSE  Makefile  Makefile.inc  Make.inc  MATLAB  PORD  README  SCILAB  src  VERSION
````

**Comment by ndellingwood** [comment]
> @ndellingwood @iyamazaki Do you have any idea what could be happening here?

I don't have any direct experience using MUMPS, thanks for the suggestions @iyamazaki !
